Apps/JobScheduler/src/Jobs/CommunicationJob.cs

```diff
@@ -0,0 +1,155 @@
+// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace Healthgateway.JobScheduler.Jobs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.Contracts;
+    using Hangfire;
+    using HealthGateway.Common.Jobs;
+    using HealthGateway.Common.Services;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Context;
+    using HealthGateway.Database.Delegates;
+    using HealthGateway.Database.Models;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc />
+    public class CommunicationJob : ICommunicationJob
+    {
+        private const int ConcurrencyTimeout = 30 * 60; // 30 minutes
+        private readonly IConfiguration configuration;
+        private readonly ILogger<CommunicationJob> logger;
+        private readonly ICommunicationDelegate communicationDelegate;
+        private readonly ICommunicationEmailDelegate commEmailDelegate;
+        private readonly IEmailQueueService emailQueueService;
+        private readonly GatewayDbContext dbContext;
+
+        private readonly int maxFetchSize;
+        private readonly string fromEmailAddressHGDonotreply;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CommunicationJob"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration to use.</param>
+        /// <param name="logger">The logger to use.</param>
+        /// <param name="communicationDelegate">The Communication delegate to use.</param>
+        /// <param name="commEmailDelegate">The Communication Email delegate to use.</param>
+        /// <param name="emailQueueService">The email queue service to use.</param>
+        /// <param name="dbContext">The db context to use.</param>
+        public CommunicationJob(IConfiguration configuration, ILogger<CommunicationJob> logger, ICommunicationDelegate communicationDelegate, ICommunicationEmailDelegate commEmailDelegate, IEmailQueueService emailQueueService, GatewayDbContext dbContext)
+        {
+            Contract.Requires((configuration != null) && (emailQueueService != null));
```

Comment:
de don't use contracts anymore. Making the objects non-nullable removes the need for them.

Reply:
removed the Contract line